src/main/java/org/alliancegenome/curation_api/jobs/executors/LoadFileExecutor.java

```diff
@@ -61,5 +69,34 @@ protected String getVersionNumber(String versionString) {
 			return versionString.substring(1);
 		return versionString;
 	}
+	
+	protected boolean validSchemaVersion(String submittedSchemaVersion, Class<?> dtoClass) {
+		
+		List<String> versionRange = apiVersionInfoService.getVersionRange(dtoClass.getAnnotation(AGRCurationSchemaVersion.class));
+		List<String> minVersionParts = new ArrayList<String>(Arrays.asList(versionRange.get(0).split("\\.")));
+		List<String> maxVersionParts = new ArrayList<String>(Arrays.asList(versionRange.get(1).split("\\.")));
+		List<String> fileVersionParts = new ArrayList<String>(Arrays.asList(submittedSchemaVersion.split("\\.")));
+		
+		while (fileVersionParts.size() < 3) { fileVersionParts.add("0"); }
+		
+		// check not lower than min version
```

Comment:
@markquintontulloch I would suggest pulling out the `Integer.parseInt(minVersionParts.get(0))` and setting it to a variable. It would make this code much easier to read.

Reply:
Plus if it doesn't parse just return false.

Reply:
Done

Reply:
I still feel confused on this one but its fine.